src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -773,6 +793,46 @@ private void handleReset(
             final long traceId = reset.trace();
             doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId);
         }
+
+        // TODO - SseEventFW
+        //                  .type = "challenge"
+        //                  .data = "{}"
+        // Received signalEx CHALLENGE from OAuth
+        private void handleSignal(
+            SignalFW signal)
+        {
+            final HttpSignalExFW httpSignalEx = httpSignalExRO.tryWrap(signal.buffer(),
+                                                                       signal.extension().offset(),
+                                                                       signal.extension().limit());
+            if (httpSignalEx != null && signal.extension().sizeof() == httpSignalExRO.limit())
+            {
+                final ListFW<HttpHeaderFW> httpHeaders = httpSignalEx.headers();
+                final Map<String, String> collectedHeaders = new HashMap<>();
```

Comment:
Rename `collectedHeaders` to `challengeHeaders`, and change to `JsonObject` to collect all non-pseudo-headers, i.e. only header names _not_ starting with `:`.

Reply:
Add `challenge` `JsonObject`, and detect `:method` during `forEach`, to set `method` property value.